runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -357,4 +354,14 @@ private static File zipDirectory(File directory) throws IOException {
       return env;
     }
   }
+
+  private static String getDefaultJavaSdkHarnessContainerUrl() {
+    String javaVersionId =
+        Float.parseFloat(System.getProperty("java.specification.version")) >= 9 ? "java11" : "java8";
```

Comment:
nit: could you make this an exact check for 8 and 11 and throw an exception for other (unsupported) versions

Reply:
I went ahead and added an Enum to Environments, and removed similar checks in DataflowRunner